### PR TITLE
Make the faker generator more unique in factories

### PR DESCRIPTION
**Pull request by outdooracorn**
Faker was generating the same "random" values causing database constraints to fail

### CODE REVIEW

**Review by toban — APPROVED** [review]
LGTM!

**Review by toban — COMMENTED** [review]
Being a bit of a php noob, I'm confused by the fact that we seem to be able to call `unique` without the parentheses, which seems to be the common way of using that method.

This also does not show up with any suggestions in my IDE, however `->unique()->text()` does. But as long as it works :)

**Comment by addshore** [comment]
> Being a bit of a php noob, I'm confused by the fact that we seem to be able to call `unique` without the parentheses, which seems to be the common way of using that method.
> 
> This also does not show up with any suggestions in my IDE, however `->unique()->text()` does. But as long as it works :)

Generally speaking, this should not work https://3v4l.org/LfigS
But I imagine that ravel is doing something (funky) here if this works

**Comment by addshore** [comment]
Per https://github.com/fzaninotto/Faker lets add the brackets

File: database/factories/InvitationFactory.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ class InvitationFactory extends Factory
     public function definition()
     {
         return [
-            'code' => $this->faker->text(10)
+            'code' => $this->faker->unique->text(20)
```
Comment (addshore) [inline database/factories/InvitationFactory.php:25]:
```suggestion
            'code' => $this->faker->unique()->text(20)
```

File: database/factories/WikiDbFactory.php [hunk 2]
```diff
@@ -22,8 +22,8 @@ class WikiDbFactory extends Factory
     public function definition()
     {
         return [
-            'name' => $this->faker->text(5),
-            'prefix' => $this->faker->text(5),
+            'name' => $this->faker->unique->text(5),
```
Comment (addshore) [inline database/factories/WikiDbFactory.php:25]:
```suggestion
            'name' => $this->faker->unique()->text(5),
```

File: database/factories/WikiDbFactory.php [hunk 3]
```diff
@@ -22,8 +22,8 @@ class WikiDbFactory extends Factory
...
+            'prefix' => $this->faker->unique->text(5),
```
Comment (addshore) [inline database/factories/WikiDbFactory.php:26]:
```suggestion
            'prefix' => $this->faker->unique()->text(5),
```

File: database/factories/WikiFactory.php [hunk 4]
```diff
@@ -21,11 +21,15 @@ class WikiFactory extends Factory
      */
     public function definition()
     {
-        $subDomainSuffix = config('wbstack.subdomain_suffix');
+        $sitename = $this->faker->unique->text(30);
```
Comment (addshore) [inline database/factories/WikiFactory.php:24]:
```suggestion
        $sitename = $this->faker->unique()->text(30);
```

File: database/factories/WikiManagerFactory.php [hunk 5]
```diff
@@ -22,8 +22,8 @@ class WikiManagerFactory extends Factory
     public function definition()
     {
         return [
-            'user_id' => $this->faker->randomNumber(),
-            'wiki_id' => $this->faker->randomNumber()
+            'user_id' => $this->faker->unique->randomNumber(),
```
Comment (addshore) [inline database/factories/WikiManagerFactory.php:25]:
```suggestion
            'user_id' => $this->faker->unique()->randomNumber(),
```

File: database/factories/WikiManagerFactory.php [hunk 6]
```diff
@@ -22,8 +22,8 @@ class WikiManagerFactory extends Factory
...
+            'wiki_id' => $this->faker->unique->randomNumber()
```
Comment (addshore) [inline database/factories/WikiManagerFactory.php:26]:
```suggestion
            'wiki_id' => $this->faker->unique()->randomNumber()
```

**Comment by outdooracorn** [comment]
> > Being a bit of a php noob, I'm confused by the fact that we seem to be able to call `unique` without the parentheses, which seems to be the common way of using that method.
> > This also does not show up with any suggestions in my IDE, however `->unique()->text()` does. But as long as it works :)
> 
> Generally speaking, this should not work https://3v4l.org/LfigS But I imagine that ravel is doing something (funky) here if this works

I imagine it's doing something similar to this https://3v4l.org/q8Ea3?

**Comment by outdooracorn** [comment]
> Per https://github.com/fzaninotto/Faker lets add the brackets

I had copied what was already used but I agree that using the `unique()` method is better as you get the benefit of the IDE helpers. Fixed.

**Comment by outdooracorn** [comment]
I'm going to wait until https://phabricator.wikimedia.org/T299335 is done before merging this otherwise it will make testing the solution harder.

**Comment by addshore** [comment]
> I'm going to wait until https://phabricator.wikimedia.org/T299335 is done before merging this otherwise it will make testing the solution harder.

I believe most of this was fixed in https://github.com/wbstack/api/commit/ba57da485a6159fadcd4860e8932dde0bbab69bf#diff-d71ef800bea77b84c84bdca0adaf30c2c76aa17e4619cae5c935829935600f40R27 ?
But on the whole I think we need to take a step back from these individual failures and look at the bigger picture to get this right

**Comment by toban** [comment]
This fixes the [uniqueness](https://github.com/wbstack/api/issues/317) problem, I've ran it locally a bunch of time without the problem happening.

Saw it again this morning here https://github.com/wbstack/api/runs/4933772817?check_suite_focus=true

```
PHPUnit 9.5.12 by Sebastian Bergmann and contributors.

..................................SS....................S........ 65 / 99 ( 65%)
...E..............................                                99 / 99 (100%)

Time: 00:05.632, Memory: 52.50 MB

There was 1 error:

1) Tests\Routes\Wiki\Managers\LogoUpdateTest::testUpdate
Illuminate\Database\QueryException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'cupiditate_.wiki.opencura.com' for key 'wikis_domain_unique' (SQL: insert into `wikis` (`sitename`, `domain`, `updated_at`, `created_at`) values (Dr. Norris Von, cupiditate_.wiki.opencura.com, 2022-01-25 08:15:22, 2022-01-25 08:15:22))

/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:712
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:672
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:502
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:454
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Processors/Processor.php:32
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2999
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:1655
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:1188
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:1153
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:994
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:314
/var/www/html/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php:242
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:317
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:261
/var/www/html/tests/Routes/Wiki/LogoUpdateTest.php:25
phpvfscomposer:///var/www/html/vendor/phpunit/phpunit/phpunit:97

Caused by
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'cupiditate_.wiki.opencura.com' for key 'wikis_domain_unique'

/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:501
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:705
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:672
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:502
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:454
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Processors/Processor.php:32
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2999
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:1655
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:1188
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:1153
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:994
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:314
/var/www/html/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php:242
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:317
/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Factories/Factory.php:261
/var/www/html/tests/Routes/Wiki/LogoUpdateTest.php:25
phpvfscomposer:///var/www/html/vendor/phpunit/phpunit/phpunit:97

ERRORS!
Tests: 99, Assertions: 263, Errors: 1, Skipped: 3.
Error: Process completed with exit code 2.
```

**Pull request merged**